### ISSUE TITLE
[Suggestion] Add support for FFmpeg

### ISSUE THREAD

**Issue by nacompllo** [visioforge/dotnet-crossplatform-sdks-samples#6]
It would be an incredible feature to support the FFmpeg library, I wanted to know if this could be in your plans to implement it in the future, this feature would make this SDK a leader in the sector without a doubt.

**Comment by visioforge** [comment]
FFMPEG as a source engine? Well, I think all FFMPEG formats and protocols are supported now.

**Comment by nacompllo** [comment]
> FFMPEG as a source engine? Well, I think all FFMPEG formats and protocols are supported now.

I mean to be able to execute any FFmpeg command, for example this one to extract the audio from a video:

  ffmpeg -i input.mkv -vn audio_only.ogg

In general, the idea I am referring to is to be able to execute any command without limitations.

**Comment by visioforge** [comment]
But you can use FFMPEG builds for various platforms, maybe with .Net wrapper.

**Comment by nacompllo** [comment]
> But you can use FFMPEG builds for various platforms, maybe with .Net wrapper.

Yes, but it really is hard work to make native libraries work on all platforms, in the Xamarin.Forms community for example there are no quality open source projects that use FFmpeg on all platforms, at most some projects on Android . In most cases it is not feasible for an individual developer to do all that work for time reasons unless it is supported by a company, so I think it would be an interesting suggestion to introduce it in this SDK in case you see it timely.

Some interesting links on how to implement native libraries in the different platforms in Xamarin:
Android: https://docs.microsoft.com/en-us/xamarin/android/platform/binding-java-library/
iOS: https://docs.microsoft.com/en-us/xamarin/ios/platform/binding-swift/walkthrough
macOS: https://docs.microsoft.com/en-us/xamarin/mac/platform/binding

**Comment by visioforge** [comment]
OK. It will be ready in 2 months :)

I think we have enough experience to do it.

**Comment by nacompllo** [comment]
> OK. It will be ready in 2 months :)
> 
> I think we have enough experience to do it.

Amazing! I am glad that you liked the suggestion, I am convinced that this SDK will be a success in the current Xamarin community and in the future .NET MAUI, good luck!

**Comment by visioforge** [comment]
Did you check our demos and problems in other issues? 

MAUI is interesting, I'm working on Uno demos right now.

**Comment by nacompllo** [comment]
> Did you check our demos and problems in other issues?
> 
> MAUI is interesting, I'm working on Uno demos right now.

Currently I have only researched this CrossPlatform Media Player repository as it is very interesting that it is geared towards so many platforms. Yes .NET MAUI looks amazing and has great support from Microsoft. Regarding Uno Platform it also has an interesting concept, when you have a demo available I will try it.

**Comment by nacompllo** [comment]
Hi, I wanted to recommend a FFmpeg library that is available for Android, iOS, macOS and tvOS in case it would be useful in this suggestion: https://github.com/tanersener/ffmpeg-kit By the way how is progress? Regards.
